core/src/subgraph/context.rs

```diff
@@ -140,6 +140,10 @@ impl<C: Blockchain, T: RuntimeHostBuilder<C>> IndexingContext<C, T> {
 
         Ok(host)
     }
+
+    pub fn causality_region_next_value(&mut self) -> CausalityRegion {
+        self.instance.causality_region_next_value()
```

Comment:
Would it make sense for this to be part of the impl of CausalityRegion? If Onchain next() is Onchain, if Offchain(n) it's Offchain(n+1) or Offchain::default() if nothing is set (ie, the first one)?